### PR TITLE
Encounters completion stats - Acute Illness

### PR DESCRIPTION
**Pull request by anvmn**
#1287

### CODE REVIEW

**Comment by anvmn** [comment]
**Initial results:**

Selection_815

**Comment by anvmn** [comment]
Results after fixes:

Selection_820

File: server/elm/src/Backend/Completion/Utils.elm [hunk 1]
```diff
@@ -59,6 +59,82 @@ nutritionMotherActivityFromMapping mapped =
             Nothing
 
 
+acuteIllnessActivityFromMapping : String -> Maybe AcuteIllnessActivity
+acuteIllnessActivityFromMapping mapped =
+    case mapped of
+        "a" ->
+            Just AcuteIllnessAcuteFindings
+
+        "b" ->
```
Comment (AronNovak) [inline server/elm/src/Backend/Completion/Utils.elm:68]: For my curiosity, where do we use these letters? Would it make debugging (like via network tab, PHP side, etc) easier if we use more expressive strings on the left side?

**Review by AronNovak — APPROVED** [review]

File: server/elm/src/Backend/Completion/Utils.elm (hunk 1 above)
Comment (anvmn) [reply server/elm/src/Backend/Completion/Utils.elm:68]: @AronNovak 
For the context:
- Data we need to present is side wide (currently, we got ~170K patients and ~500K encounters, and growing). We also need to support presentation of smaller chunks of data (HCs, Districts, Villages, etc...)
- Data needs to be filterable by period ( start date to end date).
- Data needs to be presented on fly, within reasonable period of time (up to 30 sec).
- RAM limit at Pantheon site is currently 512MB.

To support these requirements, the idea is that every patient/encounter will store it's own data locally (data is updated based on uploaded content / Jenkins tasks).
When queried for, data is generated on back-end and then attached as script for Elm APP to decode and operate with.
Also, large data sets (Global, Provinces, Districts) can not be loaded within reasonable time.
As a result, we have dedicated  CT that stores file of encoded data (updated daily).

So, the mapping we do, reduces the size of the file by factor of ~ 10.
Also, it reduces memory size we consume when loading that data before passing it to Elm App.
Mapping is 'decoded' on client side, there's no issue with debugging complexity.

**Pull request merged**